### PR TITLE
Infoblox support for Tex

### PR DESCRIPTION
**Pull request by rhafer**
This is mainly a rebase of https://github.com/crowbar/barclamp-neutron/pull/186 and https://github.com/rhafer/barclamp-neutron/pull/2 on release/tex/master.

The dhcp_relay related bits are mostly untested.

### CODE REVIEW

File: chef/cookbooks/neutron/recipes/post_install_conf.rb [hunk 1]
```diff
@@ -120,8 +120,13 @@ def mask_to_bits(mask)
   action :nothing
 end
 
+dns_options = ""
+if node[:neutron][:use_infoblox]
+    dns_options = node[:neutron][:infoblox][:ib_dnsserver].map{ |s| "--dns-nameserver #{s}" }.join(" ")
```
Comment (dirkmueller) [inline chef/cookbooks/neutron/recipes/post_install_conf.rb:119]: do we need any quoting to `#{s}` here?

**Comment by dirkmueller** [comment]
+1

**Comment by toabctl** [comment]
+1

File: chef/cookbooks/neutron/recipes/post_install_conf.rb (hunk 1 above)
Comment (rhafer) [inline chef/cookbooks/neutron/recipes/post_install_conf.rb:119]: I don't think so. It just a single IP address per `--dns-nameserver` argument.

File: chef/cookbooks/neutron/templates/default/dhcp_agent.ini.erb [hunk 2]
```diff
@@ -64,6 +73,9 @@ enable_metadata_network = <%= @enable_metadata_network %>
 
 # Location to store DHCP server config files
 # dhcp_confs = $state_path/dhcp
+<% if @ib_use_dhcp_relay -%>
+dhcp_confs = $state_path/dhcp
```
Comment (vuntz) [inline chef/cookbooks/neutron/templates/default/dhcp_agent.ini.erb:77]: That's the default, why does it need to be set?

File: chef/cookbooks/neutron/templates/default/infoblox_conditional.conf.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[
+    <%= node[:neutron][:infoblox][:ib_conditionals].map{ |c| "{\"condition\": \"#{c[:condition]}\", \"is_external\": #{c[:is_external]}, \"dns_view\": \"#{c[:dns_view]}\", \"network_view\": \"#{c[:network_view]}\", \"dhcp_members\": #{c[:dhcp_members]}, \"dns_members\": #{c[:dns_members]}, \"domain_suffix_pattern\": \"#{c[:domain_suffix_pattern]}\", \"hostname_pattern\": \"#{c[:hostname_pattern]}\", \"require_dhcp_relay\": #{node[:neutron][:infoblox][:enable_dhcp_relay]}}" }.join(",\n") %>
```
Comment (vuntz) [inline chef/cookbooks/neutron/templates/default/infoblox_conditional.conf.erb]: Can we somehow make this multi-line?

**Comment by rhafer** [comment]
> Can we somehow make this multi-line?

@vuntz: Done. 

Verified that the resulting config file still is the same (apart from the formatting).

**Comment by vuntz** [comment]
+1

**Comment by toabctl** [comment]
+1

**Comment by dirkmueller** [comment]
interestingly the CI fails pretty consistently with it..

**Comment by vuntz** [comment]
gating is green, so merging.

**Pull request merged**